javascript/ql/src/experimental/Security/CWE-347/JWTMissingSecretOrPublicKeyVerification.help

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+
+<p>The featured CodeQL query warns using of none algorithm in verify() functions imported from jsonwebtoken package developed by the auth0 organization.</p>
+
+<p>Backend JavaScript applications handling JWT could be affected by the none algorithm misconfiguration due to misusing verify() functions imported by jsonwebtoken package.
+Providing an empty string or a false value, instead of a secret or a key, enable the none algorithm to decode JWT payloads without signature verification.
+Misconfigured backend JavaScript on a production environment could be impacted by exploitation violating the integration of a JWT.</p>
+</overview>
+
+<recommendation>
+<p>
+verify() functions should use a secret or a key to decode JWT payloads.
+</p>
+<p>
+Use a a secret or a key to decode JWT payloads.
+</p>
+<p>
+</p>
+
+</recommendation>
+
+<example>
+<p>The example starts with a secret signing an object using the HS256 algorithm.
+In the second case an empty string is provided, then an undefined value, and finally a false value.
+These three misconfigued verify() functions is detected to be potentially a cybersecurity vulnerability.
```

Comment:
```suggestion
These three misconfigured calls to <code>jwt.verify()</code> can cause vulnerabilities.
```

There is no need to mention that the cases are detected, just focus on the fact that they can lead to vulnerabilities.   
I also rewrote it a bit to focus on that its a `call to jwt.verify()`. (Its not a function, its a call to a function).

Reply:
Fixed.